[WPE][Qt] Add support for building QtWPE against Qt 6

aperezdc opened the pull request:
#### 567bd6a58583b3944fc027e0d493d117aefb5337
<pre>
[WPE][Qt] Add support for building QtWPE against Qt 6
<a href="https://bugs.webkit.org/show_bug.cgi?id=270161">https://bugs.webkit.org/show_bug.cgi?id=270161</a>

Reviewed by Carlos Garcia Campos.

- Keep the existing Qt 5 based WPEQt implementation and rename:
  - Source/WebKit/UIProcess/API/wpe/qt -&gt; Source/WebKit/UIProcess/API/wpe/qt5
  - Tools/MiniBrowser/wpe/qt -&gt; Tools/MiniBrowser/wpe/qt5

- Add a new WPEQt implementation for Qt 6, based on the new WPE platform
  API, that was recently developed, not relying on WPEBackend(-fdo) anymore.

  Touch / wheel / hover / mouse / keyboard events are supported, only
  keyboard+mouse events were tested so far.

- Polish UI: Add navigation buttons (back|forward|reload) / URL bar for
  Qt 6 MiniBrowser

- Make the WebKitWebView ref-counted in WPEQtView, by wrapping in
  GRefPtr (needs some moc related plumbing to allow for config.h
  inclusion, which is a pre-requisite to get wtf/ headers includable).

Tested using:
build-webkit --wpe --debug --cmakeargs &quot;-DUSE_QT6=ON|OFF&quot;
run-qt-wpe-minibrowser --debug <a href="https://www.igalia.com">https://www.igalia.com</a>

Covered by existing tests, no change in default behavior.

* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/UIProcess/API/wpe/qt5/CMakeLists.txt: Renamed from Source/WebKit/UIProcess/API/wpe/qt/CMakeLists.txt.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQmlExtensionPlugin.cpp: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQmlExtensionPlugin.cpp.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQmlExtensionPlugin.h: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQmlExtensionPlugin.h.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtView.cpp: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtView.cpp.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtView.h: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtView.h.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewBackend.cpp: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewBackend.cpp.
(WPEQtViewBackend::texture):
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewBackend.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewBackend.h.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewLoadRequest.cpp: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequest.cpp.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewLoadRequest.h: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequest.h.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewLoadRequestPrivate.h: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequestPrivate.h.
* Source/WebKit/UIProcess/API/wpe/qt5/qmldir: Copied from Source/WebKit/UIProcess/API/wpe/qt/qmldir.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.cpp: Added.
(wpeDisplayQtQuickConnect):
(wpeDisplayQtQuickCreateView):
(wpeDisplayQtQuickGetEGLDisplay):
(wpeDisplayQtQuickGetDRMDevice):
(wpeDisplayQtQuickGetDRMRenderNode):
(wpe_display_qtquick_class_init):
(wpe_display_qtquick_new):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.h: Added.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQmlExtensionPlugin.cpp: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQmlExtensionPlugin.cpp.
(WPEQmlExtensionPlugin::registerTypes):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQmlExtensionPlugin.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQmlExtensionPlugin.h.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtView.cpp.
(WPEQtView::WPEQtView):
(WPEQtView::~WPEQtView):
(WPEQtView::geometryChange):
(WPEQtView::configureWindow):
(WPEQtView::createWebView):
(WPEQtView::notifyUrlChangedCallback):
(WPEQtView::notifyTitleChangedCallback):
(WPEQtView::notifyLoadProgressCallback):
(WPEQtView::notifyLoadChangedCallback):
(WPEQtView::notifyLoadFailedCallback):
(WPEQtView::didUpdateScene):
(WPEQtView::updatePaintNode):
(WPEQtView::url const):
(WPEQtView::setUrl):
(WPEQtView::loadProgress const):
(WPEQtView::title const):
(WPEQtView::canGoBack const):
(WPEQtView::isLoading const):
(WPEQtView::canGoForward const):
(WPEQtView::goBack):
(WPEQtView::goForward):
(WPEQtView::reload):
(WPEQtView::stop):
(WPEQtView::loadHtml):
(JavascriptCallbackData::JavascriptCallbackData):
(jsAsyncReadyCallback):
(WPEQtView::runJavaScript):
(WPEQtView::mousePressEvent):
(WPEQtView::mouseMoveEvent):
(WPEQtView::mouseReleaseEvent):
(WPEQtView::hoverEnterEvent):
(WPEQtView::hoverLeaveEvent):
(WPEQtView::hoverMoveEvent):
(WPEQtView::wheelEvent):
(WPEQtView::keyPressEvent):
(WPEQtView::keyReleaseEvent):
(WPEQtView::touchEvent):
(WPEQtView::webView const):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtView.h.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtViewLoadRequest.cpp: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequest.cpp.
(WPEQtViewLoadRequest::WPEQtViewLoadRequest):
(WPEQtViewLoadRequest::~WPEQtViewLoadRequest):
(WPEQtViewLoadRequest::url const):
(WPEQtViewLoadRequest::status const):
(WPEQtViewLoadRequest::errorString const):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtViewLoadRequest.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequest.h.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtViewLoadRequestPrivate.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequestPrivate.h.
(WPEQtViewLoadRequestPrivate::WPEQtViewLoadRequestPrivate):
(WPEQtViewLoadRequestPrivate::~WPEQtViewLoadRequestPrivate):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEViewQtQuick.cpp: Added.
(wpeViewQtQuickDispose):
(wpeViewQtQuickRenderBuffer):
(wpeViewQtQuickResize):
(wpe_view_qtquick_class_init):
(wpe_view_qtquick_new):
(wpe_view_qtquick_initialize_rendering):
(wpe_view_qtquick_render_buffer_to_texture):
(wpe_view_qtquick_did_update_scene):
(buttonFromEvent):
(mouseModifiersFromEvent):
(keyboardModifiersFromEvent):
(modifiersFromEvent):
(wpe_view_dispatch_mouse_press_event):
(wpe_view_dispatch_mouse_move_event):
(wpe_view_dispatch_mouse_release_event):
(wpe_view_dispatch_wheel_event):
(wpe_view_dispatch_hover_enter_event):
(wpe_view_dispatch_hover_move_event):
(wpe_view_dispatch_hover_leave_event):
(wpe_view_dispatch_key_press_event):
(wpe_view_dispatch_key_release_event):
(wpe_view_dispatch_touch_event):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEViewQtQuick.h: Added.
* Source/WebKit/UIProcess/API/wpe/qt6/qmldir: Renamed from Source/WebKit/UIProcess/API/wpe/qt/qmldir.
* Source/cmake/OptionsWPE.cmake:
* Tools/MiniBrowser/wpe/CMakeLists.txt:
* Tools/MiniBrowser/wpe/qt/CMakeLists.txt: Removed.
* Tools/MiniBrowser/wpe/qt5/CMakeLists.txt: Added.
* Tools/MiniBrowser/wpe/qt5/main.cpp: Copied from Tools/MiniBrowser/wpe/qt/main.cpp.
* Tools/MiniBrowser/wpe/qt5/main.qml: Renamed from Tools/MiniBrowser/wpe/qt/main.qml.
* Tools/MiniBrowser/wpe/qt5/qml.qrc: Copied from Tools/MiniBrowser/wpe/qt/qml.qrc.
* Tools/MiniBrowser/wpe/qt6/CMakeLists.txt: Added.
* Tools/MiniBrowser/wpe/qt6/main.cpp: Renamed from Tools/MiniBrowser/wpe/qt/main.cpp.
(main):
* Tools/MiniBrowser/wpe/qt6/main.qml: Added.
* Tools/MiniBrowser/wpe/qt6/qml.qrc: Renamed from Tools/MiniBrowser/wpe/qt/qml.qrc.
* Tools/Scripts/run-qt-wpe-minibrowser:
* Tools/Scripts/webkitdirs.pm:
(builtDylibPathForName):
* Tools/Scripts/webkitpy/style/checker.py:
* Tools/TestWebKitAPI/Tests/WPEQt/WPEQtTest.cpp:
* Tools/TestWebKitAPI/Tests/WPEQt/WPEQtTest.h:
* Tools/TestWebKitAPI/glib/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/279129@main">https://commits.webkit.org/279129@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c701f51b2c5124ec6a137b4ba59a77cd9a3f8f3e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52589 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31923 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5015 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55863 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3312 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/54894 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/38496 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3013 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/42727 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2120 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54687 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29610 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45387 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23822 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/52433 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/26754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2668 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1471 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/45940 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/48638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2821 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57459 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/52099 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27724 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/2839 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50122 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/28950 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45503 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/49391 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11482 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/29863 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/64405 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28700 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12209 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp:
```diff
@@ -0,0 +1,542 @@
+        g_signal_handlers_disconnect_by_func(m_webView.get(), reinterpret_cast<gpointer>(notifyTitleChangedCallback), this);
+        g_signal_handlers_disconnect_by_func(m_webView.get(), reinterpret_cast<gpointer>(notifyLoadChangedCallback), this);
+        g_signal_handlers_disconnect_by_func(m_webView.get(), reinterpret_cast<gpointer>(notifyLoadFailedCallback), this);
+        g_signal_handlers_disconnect_by_func(m_webView.get(), reinterpret_cast<gpointer>(notifyLoadProgressCallback), this);
```
You can probably use g_signal_handlers_disconnect_by_data().

nikolaszimmermann replied:
Great, fixed.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEViewQtQuick.cpp:
```diff
@@ -0,0 +1,453 @@
+static void wpeViewQtQuickConstructed(GObject* object)
+{
+    G_OBJECT_CLASS(wpe_view_qtquick_parent_class)->constructed(object);
+}
```
This is not needed

nikolaszimmermann replied:
Fixed.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEViewQtQuick.cpp:
```diff
@@ -0,0 +1,453 @@
+{
+    auto* priv = WPE_VIEW_QTQUICK(object)->priv;
+    if (auto* glFunctions = priv->context ? priv->context->functions() : nullptr)
+        glFunctions->glDeleteTextures(1, &priv->textureId);
```
dispose can be called multiple times, you should set textureId = 0 and check it before deleting it

nikolaszimmermann replied:
ic - need to learn more about GLib, I wasn't aware of that.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEViewQtQuick.h:
```diff
@@ -0,0 +1,67 @@
+#define WPE_TYPE_VIEW_QTQUICK (wpe_view_qtquick_get_type())
+WPE_API G_DECLARE_FINAL_TYPE (WPEViewQtQuick, wpe_view_qtquick, WPE, VIEW_QTQUICK, WPEView)
+
+WPE_API WPEView *wpe_view_qtquick_new                      (WPEDisplayQtQuick *display);
```
This shouldn't be public, I guess

nikolaszimmermann replied:
I guess removing from both lines WPE_API you mean? I'll check and fix.

nikolaszimmermann replied:
Removed both.

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPEExtensions.h:
```diff
@@ -25,5 +25,5 @@
 #pragma once
 
-void wpeEnsureExtensionPointsRegistered();
+WTF_EXPORT_DECLARATION void wpeEnsureExtensionPointsRegistered();
```
This shouldn't be needed.

nikolaszimmermann replied:
Right, it's no longer needed.

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPEView.h:
```diff
@@ -96,11 +96,13 @@ typedef enum {
 /**
  * WPEViewError:
  * @WPE_VIEW_ERROR_RENDER_FAILED: Failed to render
+ * @WPE_VIEW_ERROR_INITIALIZATION_FAILED: Failed to initialize rendering
```
This is weird, because in the base api there's no api to initialize rendering, I guess we can just use render error in the end.

nikolaszimmermann replied:
You're right -- that category would only be used by the internal QtQuick view/display - makes no sense to introduce that type - will remove it.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.h:
```diff
@@ -0,0 +1,41 @@
+#define WPE_TYPE_DISPLAY_QTQUICK (wpe_display_qtquick_get_type())
+WPE_API G_DECLARE_FINAL_TYPE (WPEDisplayQtQuick, wpe_display_qtquick, WPE, DISPLAY_QTQUICK, WPEDisplay)
+
+WPE_API WPEDisplay *wpe_display_qtquick_new (void);
```
I guess we are not exposing this in the qt api, so this doesn't need to be public.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp:
```diff
@@ -0,0 +1,537 @@
+        return;
+
+    if (auto* wpeView = webkit_web_view_get_wpe_view(m_webView.get()))
+        wpe_view_resize(wpeView, m_size.width(), m_size.height());
```
Is this expected to work with the old api too? Because when using the new api `webkit_web_view_get_wpe_view` should always return a non-null pointer.

nikolaszimmermann replied:
Nope only new API, I will adapt - wasn't sure about that.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp:
```diff
@@ -0,0 +1,537 @@
+    RELEASE_ASSERT(!m_webView);
+
+    GUniqueOutPtr<GError> error;
+    auto* wpeDisplay = wpe_display_qtquick_new();
+    if (!wpe_display_connect(wpeDisplay, &error.outPtr())) {
+        g_warning("Failed to initialize WPE display: %s", error->message);
+        return;
+    }
```
Are multiple views supported? Because in that case they should share the same display, not create one per view.

nikolaszimmermann replied:
Nope, multiple views are not supported, one view per QML web view.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp:
```diff
@@ -0,0 +1,537 @@
+    }
+
+    auto settings = adoptGRef(webkit_settings_new_with_settings("enable-developer-extras", TRUE,
+        "enable-webgl", TRUE, "enable-mediasource", TRUE, nullptr));
```
webgl and mediasource are enabled by default.

nikolaszimmermann replied:
Good point, will adapt.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp:
```diff
@@ -0,0 +1,537 @@
+        "settings", settings.get(), nullptr)));
+
+    auto* wpeView = webkit_web_view_get_wpe_view(m_webView.get());
+    RELEASE_ASSERT(wpeView);
```
I don't think the assert is needed, you have just created the view with a WPEDisplay.

nikolaszimmermann replied:
Ok, I'll get rid of them.

nikolaszimmermann committed:
[WPE][Qt] Add support for building QtWPE against Qt 6
https://bugs.webkit.org/show_bug.cgi?id=270161

Reviewed by Carlos Garcia Campos.

- Keep the existing Qt 5 based WPEQt implementation and rename:
  - Source/WebKit/UIProcess/API/wpe/qt -> Source/WebKit/UIProcess/API/wpe/qt5
  - Tools/MiniBrowser/wpe/qt -> Tools/MiniBrowser/wpe/qt5

- Add a new WPEQt implementation for Qt 6, based on the new WPE platform
  API, that was recently developed, not relying on WPEBackend(-fdo) anymore.

  Touch / wheel / hover / mouse / keyboard events are supported, only
  keyboard+mouse events were tested so far.

- Polish UI: Add navigation buttons (back|forward|reload) / URL bar for
  Qt 6 MiniBrowser

- Make the WebKitWebView ref-counted in WPEQtView, by wrapping in
  GRefPtr (needs some moc related plumbing to allow for config.h
  inclusion, which is a pre-requisite to get wtf/ headers includable).

Tested using:
build-webkit --wpe --debug --cmakeargs "-DUSE_QT6=ON|OFF"
run-qt-wpe-minibrowser --debug https://www.igalia.com

Covered by existing tests, no change in default behavior.

* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/UIProcess/API/wpe/qt5/CMakeLists.txt: Renamed from Source/WebKit/UIProcess/API/wpe/qt/CMakeLists.txt.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQmlExtensionPlugin.cpp: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQmlExtensionPlugin.cpp.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQmlExtensionPlugin.h: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQmlExtensionPlugin.h.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtView.cpp: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtView.cpp.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtView.h: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtView.h.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewBackend.cpp: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewBackend.cpp.
(WPEQtViewBackend::texture):
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewBackend.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewBackend.h.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewLoadRequest.cpp: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequest.cpp.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewLoadRequest.h: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequest.h.
* Source/WebKit/UIProcess/API/wpe/qt5/WPEQtViewLoadRequestPrivate.h: Copied from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequestPrivate.h.
* Source/WebKit/UIProcess/API/wpe/qt5/qmldir: Copied from Source/WebKit/UIProcess/API/wpe/qt/qmldir.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.cpp: Added.
(wpeDisplayQtQuickConnect):
(wpeDisplayQtQuickCreateView):
(wpeDisplayQtQuickGetEGLDisplay):
(wpeDisplayQtQuickGetDRMDevice):
(wpeDisplayQtQuickGetDRMRenderNode):
(wpe_display_qtquick_class_init):
(wpe_display_qtquick_new):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.h: Added.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQmlExtensionPlugin.cpp: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQmlExtensionPlugin.cpp.
(WPEQmlExtensionPlugin::registerTypes):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQmlExtensionPlugin.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQmlExtensionPlugin.h.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtView.cpp.
(WPEQtView::WPEQtView):
(WPEQtView::~WPEQtView):
(WPEQtView::geometryChange):
(WPEQtView::configureWindow):
(WPEQtView::createWebView):
(WPEQtView::notifyUrlChangedCallback):
(WPEQtView::notifyTitleChangedCallback):
(WPEQtView::notifyLoadProgressCallback):
(WPEQtView::notifyLoadChangedCallback):
(WPEQtView::notifyLoadFailedCallback):
(WPEQtView::didUpdateScene):
(WPEQtView::updatePaintNode):
(WPEQtView::url const):
(WPEQtView::setUrl):
(WPEQtView::loadProgress const):
(WPEQtView::title const):
(WPEQtView::canGoBack const):
(WPEQtView::isLoading const):
(WPEQtView::canGoForward const):
(WPEQtView::goBack):
(WPEQtView::goForward):
(WPEQtView::reload):
(WPEQtView::stop):
(WPEQtView::loadHtml):
(JavascriptCallbackData::JavascriptCallbackData):
(jsAsyncReadyCallback):
(WPEQtView::runJavaScript):
(WPEQtView::mousePressEvent):
(WPEQtView::mouseMoveEvent):
(WPEQtView::mouseReleaseEvent):
(WPEQtView::hoverEnterEvent):
(WPEQtView::hoverLeaveEvent):
(WPEQtView::hoverMoveEvent):
(WPEQtView::wheelEvent):
(WPEQtView::keyPressEvent):
(WPEQtView::keyReleaseEvent):
(WPEQtView::touchEvent):
(WPEQtView::webView const):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtView.h.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtViewLoadRequest.cpp: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequest.cpp.
(WPEQtViewLoadRequest::WPEQtViewLoadRequest):
(WPEQtViewLoadRequest::~WPEQtViewLoadRequest):
(WPEQtViewLoadRequest::url const):
(WPEQtViewLoadRequest::status const):
(WPEQtViewLoadRequest::errorString const):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtViewLoadRequest.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequest.h.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEQtViewLoadRequestPrivate.h: Renamed from Source/WebKit/UIProcess/API/wpe/qt/WPEQtViewLoadRequestPrivate.h.
(WPEQtViewLoadRequestPrivate::WPEQtViewLoadRequestPrivate):
(WPEQtViewLoadRequestPrivate::~WPEQtViewLoadRequestPrivate):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEViewQtQuick.cpp: Added.
(wpeViewQtQuickDispose):
(wpeViewQtQuickRenderBuffer):
(wpeViewQtQuickResize):
(wpe_view_qtquick_class_init):
(wpe_view_qtquick_new):
(wpe_view_qtquick_initialize_rendering):
(wpe_view_qtquick_render_buffer_to_texture):
(wpe_view_qtquick_did_update_scene):
(buttonFromEvent):
(mouseModifiersFromEvent):
(keyboardModifiersFromEvent):
(modifiersFromEvent):
(wpe_view_dispatch_mouse_press_event):
(wpe_view_dispatch_mouse_move_event):
(wpe_view_dispatch_mouse_release_event):
(wpe_view_dispatch_wheel_event):
(wpe_view_dispatch_hover_enter_event):
(wpe_view_dispatch_hover_move_event):
(wpe_view_dispatch_hover_leave_event):
(wpe_view_dispatch_key_press_event):
(wpe_view_dispatch_key_release_event):
(wpe_view_dispatch_touch_event):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEViewQtQuick.h: Added.
* Source/WebKit/UIProcess/API/wpe/qt6/qmldir: Renamed from Source/WebKit/UIProcess/API/wpe/qt/qmldir.
* Source/cmake/OptionsWPE.cmake:
* Tools/MiniBrowser/wpe/CMakeLists.txt:
* Tools/MiniBrowser/wpe/qt/CMakeLists.txt: Removed.
* Tools/MiniBrowser/wpe/qt5/CMakeLists.txt: Added.
* Tools/MiniBrowser/wpe/qt5/main.cpp: Copied from Tools/MiniBrowser/wpe/qt/main.cpp.
* Tools/MiniBrowser/wpe/qt5/main.qml: Renamed from Tools/MiniBrowser/wpe/qt/main.qml.
* Tools/MiniBrowser/wpe/qt5/qml.qrc: Copied from Tools/MiniBrowser/wpe/qt/qml.qrc.
* Tools/MiniBrowser/wpe/qt6/CMakeLists.txt: Added.
* Tools/MiniBrowser/wpe/qt6/main.cpp: Renamed from Tools/MiniBrowser/wpe/qt/main.cpp.
(main):
* Tools/MiniBrowser/wpe/qt6/main.qml: Added.
* Tools/MiniBrowser/wpe/qt6/qml.qrc: Renamed from Tools/MiniBrowser/wpe/qt/qml.qrc.
* Tools/Scripts/run-qt-wpe-minibrowser:
* Tools/Scripts/webkitdirs.pm:
(builtDylibPathForName):
* Tools/Scripts/webkitpy/style/checker.py:
* Tools/TestWebKitAPI/Tests/WPEQt/WPEQtTest.cpp:
* Tools/TestWebKitAPI/Tests/WPEQt/WPEQtTest.h:
* Tools/TestWebKitAPI/glib/CMakeLists.txt:

Canonical link: https://commits.webkit.org/279129@main
